### PR TITLE
nix: refatored how shell & tools are provided

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

**Review by semanticphilosopher — APPROVED** [review]
All seems reasonable to me

**Comment by coot** [comment]
@semanticphilosopher I am stealing ideas from the `ouroboros-consensus` repo, without making it too hard to understand what's going on.

**Comment by coot** [comment]
@semanticphilosopher you need to review as @njd42, we only have your latter account in the @IntersectMBO/ouroboros-network-maintainers.

File: nix/ouroboros-network.nix [hunk 1]
```diff
@@ -29,105 +29,104 @@ let
...
+        default = { compiler-nix-name = defaultCompiler; };
+      };
+
     #
     # CHaP
     # ----
 
     # CHaP input map, so we can find CHaP packages (needs to be more
-    # recent than the index-state we set!). Can be updated with
+    # recenbt than the index-state we set!). Can be updated with
```
Comment (bolt12) [inline nix/ouroboros-network.nix:95]: typo

Comment (coot) [reply nix/ouroboros-network.nix]: Thanks, addressed.

**Review by bolt12 — APPROVED** [review]

**Pull request merged**